packages/headless/src/controllers/facets/_common/facet-id-generator.ts

```diff
@@ -0,0 +1,50 @@
+import {FacetType} from '../../../features/facets/facet-api/request';
+import {AnyFacetSetState} from '../../../features/facets/generic/interfaces/generic-facet-state';
+
+export interface Logger {
+  warn: (message: string) => void;
+}
+
+export interface FacetIdConfig {
+  type: FacetType;
+  field: string;
+  state: AnyFacetSetState;
+}
+
+export function generateFacetId(config: FacetIdConfig, logger: Logger) {
+  const {type, field, state} = config;
+
+  const prefix = `${type}_${field}_`;
+  const id = calculateId(prefix, state);
+
+  logWarningMessageIfNeeded(id, field, logger);
+
+  return `${prefix}${id}`;
```

Comment:
Let's discuss the format of the id, since we may display it publicly in the url. The current format is:

```
`${type}_${field}_{number}`
```

The type can potentially be removed if instead of passing the facet set to the function, we pass the entire state object and check all 4 facet sets.

Another thing could be to not `_0` for the first id and only use the field. The second field would then use `_0` and so on.

Reply:
Thinking back on this following scrum discussion, I think I would remove the prefix mechanism.
And be a bit more lax on the strictness check for collision against same field used for different facet types, at least for now.

Numeric facet cannot be used as "specific" facets, and vice versa. In those case, the index will simply always return an empty (useless) facet.



Also, I would not add `_0` for the first id.

Generally speaking, we can assume that most of the time, there will be no collision, and facets fields will be unique for the vast majority of end users.

So, I would vote to optimize for that instead, making the identifier more human friendly, and more usable as is for an URL representation.

Reply:
I would also start at `_1` after the first field, just a little detail

Reply:
I prefer `_0` its a bit more predictable since the structure of the id is always 100% the same.

Reply:
Indeed, I used `_0` for the predictability. However, the first id will now just be the field, so the form can no longer be consistent. I think it is fair that the second id ends with `_1`, but it is a small detail either way,